### PR TITLE
Bump mysqlclient to 2.2

### PR DESCRIPTION
**Pull request by srittau**
The first commit just changes `Any` instances to `Incomplete`. The following commit(s) contain the changes for 2.2.

Closes: #10347

### CODE REVIEW

File: stubs/mysqlclient/MySQLdb/cursors.pyi [hunk 1]
```diff
@@ -34,26 +36,29 @@ class BaseCursor:
     def setinputsizes(self, *args) -> None: ...
     def setoutputsizes(self, *args) -> None: ...
     def execute(self, query, args: Incomplete | None = None): ...
-    def executemany(self, query: str, args: list[Any]) -> int: ...
+    def mogrify(
+        self, query: str | bytes, args: dict[str, _Literal] | dict[bytes, _Literal] | Iterable[_Literal] | None = None
```
Comment (srittau) [inline stubs/mysqlclient/MySQLdb/cursors.pyi:40]: Does `dict[str, _Literal]` work despite the invariance of `dict`? `_Literal` contains `Any`, but also other types. Should we use `dict[str, Any]`, instead?

File: stubs/mysqlclient/MySQLdb/connections.pyi [hunk 2]
```diff
@@ -16,16 +17,21 @@ from ._exceptions import (
     Warning as Warning,
 )
 
-re_numeric_part: Any
+# Any kind of object that can be passed to Connection.literal().
+# This allowed types depend on the defined encoders, but the following
```
Comment (AlexWaygood) [inline stubs/mysqlclient/MySQLdb/connections.pyi:21]:
```suggestion
# The allowed types depend on the defined encoders, but the following
```

File: stubs/mysqlclient/MySQLdb/connections.pyi [hunk 3]
```diff
@@ -16,16 +17,21 @@ from ._exceptions import (
...
+# types are always allowed.
+_Literal: TypeAlias = str | bytearray | bytes | tuple[_Literal] | list[_Literal] | Any
```
Comment (AlexWaygood) [inline stubs/mysqlclient/MySQLdb/connections.pyi:23]:
```suggestion
_Literal: TypeAlias = str | bytearray | bytes | tuple[_Literal, ...] | list[_Literal] | Any
```

File: stubs/mysqlclient/MySQLdb/connections.pyi [hunk 4]
```diff
@@ -16,16 +17,21 @@ from ._exceptions import (
...
+
+re_numeric_part: Incomplete
```
Comment (AlexWaygood) [inline stubs/mysqlclient/MySQLdb/connections.pyi:25]:
```suggestion
re_numeric_part: Pattern[str]
```

File: stubs/mysqlclient/MySQLdb/cursors.pyi [hunk 5]
```diff
@@ -1,7 +1,9 @@
 from _typeshed import Incomplete
-from typing import Any
+from collections.abc import Iterable
 
-RE_INSERT_VALUES: Any
+from .connections import _Literal
+
+RE_INSERT_VALUES: Incomplete
```
Comment (AlexWaygood) [inline stubs/mysqlclient/MySQLdb/cursors.pyi:6]:
```suggestion
RE_INSERT_VALUES: Pattern[str]
```

File: stubs/mysqlclient/MySQLdb/cursors.pyi [hunk 6]
```diff
@@ -34,26 +36,29 @@ class BaseCursor:
     def setinputsizes(self, *args) -> None: ...
     def setoutputsizes(self, *args) -> None: ...
     def execute(self, query, args: Incomplete | None = None): ...
-    def executemany(self, query: str, args: list[Any]) -> int: ...
+    def mogrify(
+        self, query: str | bytes, args: dict[str, _Literal] | dict[bytes, _Literal] | Iterable[_Literal] | None = None
+    ) -> str: ...
+    def executemany(self, query: str, args: list[Incomplete]) -> int: ...
```
Comment (AlexWaygood) [inline stubs/mysqlclient/MySQLdb/cursors.pyi:42]: It's only ever iterated over, so I think any `Iterable` will do

```suggestion
    def executemany(self, query: str, args: Iterable[Incomplete]) -> int: ...
```

File: stubs/mysqlclient/MySQLdb/cursors.pyi [hunk 7]
```diff
@@ -34,26 +36,29 @@ class BaseCursor:
...
 class CursorUseResultMixIn:
-    rownumber: Any
+    rownumber: Incomplete
     def fetchone(self): ...
     def fetchmany(self, size: Incomplete | None = None): ...
     def fetchall(self): ...
     def __iter__(self): ...
     def next(self): ...
-    __next__: Any
+    __next__: Incomplete
```
Comment (AlexWaygood) [inline stubs/mysqlclient/MySQLdb/cursors.pyi:61]:
```suggestion
    __next__ = next
```

File: stubs/mysqlclient/MySQLdb/cursors.pyi (hunk 1 above)
Comment (AlexWaygood) [reply stubs/mysqlclient/MySQLdb/cursors.pyi:40]: I think what you've got _should_ be fine. If it causes issues for somebody, we'll probably get a complaint about it, and we can change it then ;)

**Review by AlexWaygood — APPROVED** [review]
LGTM

**Pull request merged**